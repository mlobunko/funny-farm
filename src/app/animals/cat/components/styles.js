import styled from 'styled-components';
import imageAnimal from '../images/animal.svg';
import imageAnimalChat from '../images/animal-chat.svg';

export const StyledCatComponent = styled.div.attrs({
  'aria-label': 'play cat sound button'
})`
  width: 20vw;
  background: ${props =>
    props.chat
      ? `url(${imageAnimalChat}) no-repeat center center`
      : `url(${imageAnimal}) no-repeat center center`};
  background-size: contain;
  touch-action: none;
`;
