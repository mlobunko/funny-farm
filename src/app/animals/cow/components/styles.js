import styled from 'styled-components';
import PropTypes from 'prop-types';

import imageAnimal from '../images/animal.svg';
import imageAnimalChat from '../images/animal-chat.svg';

export const StyledCowComponent = styled.div.attrs({
  'aria-label': 'play cow sound button'
})`
  width: 20vw;
  background: ${props =>
    props.chat
      ? `url(${imageAnimalChat}) no-repeat center center`
      : `url(${imageAnimal}) no-repeat center center`};
  background-size: contain;
  touch-action: none;
`;

StyledCowComponent.displayName = 'StyledCowComponent';

StyledCowComponent.propTypes = {
  chat: PropTypes.bool
};
