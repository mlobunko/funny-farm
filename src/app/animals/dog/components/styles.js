import styled from 'styled-components';
import PropTypes from 'prop-types';

import imageAnimal from '../images/animal.svg';
import imageAnimalChat from '../images/animal-chat.svg';

export const StyledDogComponent = styled.div.attrs({
  'aria-label': 'play dog sound button'
})`
  width: 20vw;
  background: ${props =>
    props.chat
      ? `url(${imageAnimalChat}) no-repeat center center`
      : `url(${imageAnimal}) no-repeat center center`};
  background-size: contain;
  touch-action: none;
`;

StyledDogComponent.displayName = 'StyledDogComponent';

StyledDogComponent.propTypes = {
  chat: PropTypes.bool
};
