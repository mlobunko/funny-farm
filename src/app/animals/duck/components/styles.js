import styled from 'styled-components';
import PropTypes from 'prop-types';

import imageAnimal from '../images/animal.svg';
import imageAnimalChat from '../images/animal-chat.svg';

export const StyledDuckComponent = styled.div.attrs({
  'aria-label': 'play duck sound button'
})`
  width: 20vw;
  background: ${props =>
    props.chat
      ? `url(${imageAnimalChat}) no-repeat center center`
      : `url(${imageAnimal}) no-repeat center center`};
  background-size: contain;
  touch-action: none;
`;

StyledDuckComponent.displayName = 'StyledDuckComponent';

StyledDuckComponent.propTypes = {
  chat: PropTypes.bool
};
