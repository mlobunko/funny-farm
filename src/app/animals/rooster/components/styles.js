import styled from 'styled-components';
import imageAnimal from '../images/animal.svg';
import imageAnimalChat from '../images/animal-chat.svg';

export const StyledRoosterComponent = styled.div.attrs({
  'aria-label': 'play rooster sound button'
})`
  width: 20vw;
  background: ${props =>
    props.chat
      ? `url(${imageAnimalChat}) no-repeat center center`
      : `url(${imageAnimal}) no-repeat center center`};
  background-size: contain;
  touch-action: none;
`;
