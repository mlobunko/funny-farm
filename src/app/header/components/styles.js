import styled from 'styled-components';
import soundOn from '../images/sound-on.svg';
import soundOff from '../images/sound-off.svg';

export const StyledHeaderComponent = styled.div`
  display: flex;
  justify-content: space-between;
  height: 6rem;
  padding-top: 2rem;

  @media (min-height: 600px) {
    height: 8rem;
  }
  @media (min-width: 2000px) {
    height: 15rem;
  }
`;

export const StyledLeft = styled.div`
  flex: 1 0;
  width: 1rem;
  height: 100%;
`;

export const StyledH1 = styled.h1`
  font-size: 3rem;
  margin: 0;
  padding: 0;
  font-family: LuckiestGuy;
  letter-spacing: 2px;
  color: white;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;

  @media (min-height: 600px) {
    text-shadow: -2px 0 black, 0 2px black, 2px 0 black, 0 -2px black;
    font-size: 5rem;
  }

  @media (min-width: 2000px) {
    font-size: 9rem;
  }
`;

export const StyledRight = styled.div`
  height: 100%;
  flex: 1 0;
  display: flex;
  justify-content: flex-end;
`;

export const StyledMusicIcon = styled.div.attrs({
  'aria-label': 'toggle play music'
})`
  width: 3rem;
  touch-action: none;
  background: ${props =>
    props.isMusicOn
      ? `url(${soundOn}) no-repeat center center`
      : `url(${soundOff}) no-repeat center center`};
  background-size: contain;
  margin-right: 2rem;
  cursor: pointer;
  @media (min-width: 900px) {
    width: 5rem;
  }

  @media (min-width: 2000px) {
    width: 8rem;
    margin-right: 4rem;
  }
`;

StyledMusicIcon.displayName = 'StyledMusicIcon';
