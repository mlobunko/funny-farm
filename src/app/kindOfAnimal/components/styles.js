import styled, { keyframes } from 'styled-components';

const commonStyles = `
  height: 9vh;
  text-align: center;
  font-family: LuckiestGuy;
  letter-spacing: 2px;
  color: white;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
`;

export const StyledKind = styled.div`
  ${commonStyles};
  font-size: 2.5rem;

  @media (min-height: 600px) {
    font-size: 4rem;
    text-shadow: -2px 0 black, 0 2px black, 2px 0 black, 0 -2px black;
  }

  @media (min-width: 2000px) {
    font-size: 7rem;
  }
`;

const blinkingText = keyframes`
  0% {
    opacity: 1;
  }
  50% {
    opacity: 0.5;
  }
  100% {
    opacity: 1;
  }
`;

export const StyledAnimation = styled.div`
  ${commonStyles};
  font-size: 1.5rem;
  animation-name: ${blinkingText};
  animation-duration: 1s;
  animation-iteration-count: infinite;
  opacity: 1;

  @media (min-height: 600px) {
    font-size: 2.3rem;
  }

  @media (min-width: 2000px) {
    font-size: 4rem;
  }
`;
