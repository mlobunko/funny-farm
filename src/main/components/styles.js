import styled, { keyframes, injectGlobal } from 'styled-components';
import PropTypes from 'prop-types';

import imageRotate from '../images/rotate.svg';
import imageBackground from '../images/bg.jpg';
import fontLuckiestGuy from '../fonts/LuckiestGuy.ttf';

injectGlobal`
  html {
    font-size: 62.5%;
  }
`;

export const StyledAnimals = styled.div`
  min-height: 70vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`;
StyledAnimals.displayName = 'StyledAnimals';

export const StyledContentComponent = styled.div`
  min-height: 100vh;
  min-width: 100vw;
  background: url(${imageBackground}) no-repeat center center;
  background-size: cover;
  @font-face {
    font-family: LuckiestGuy;
    src: url(${fontLuckiestGuy});
  }
`;
StyledContentComponent.displayName = 'StyledContentComponent';

export const StyledImageRotate = styled.div`
  width: 15rem;
  height: 15rem;
  background: url(${imageRotate}) no-repeat center center;
  background-size: cover;
`;
StyledImageRotate.displayName = 'StyledImageRotate';

const rotate90deg = keyframes`
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(-30deg);
  }
  100% {
    transform: rotate(0deg);
  }
`;

export const StyledRotate = styled.div`
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  animation: ${rotate90deg} 1.5s;
  animation-iteration-count: infinite;
`;
StyledRotate.displayName = 'StyledRotate';

export const StyledRow = styled.div`
  margin-top: ${props => props.top && '1.6rem'};
  height: 18vh;
  display: flex;
  justify-content: ${props => (props.top ? 'space-evenly' : 'center')};
`;
StyledRow.displayName = 'StyledRow';

StyledRow.propTypes = {
  top: PropTypes.bool
};
