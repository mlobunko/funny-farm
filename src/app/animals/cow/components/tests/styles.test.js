import React from 'react';
import { shallow } from 'enzyme';
import 'enzyme-to-json';

import { StyledCowComponent } from '../styles';

describe('StyledCatComponent', () => {
  it('renders animal', () => {
    const wrapper = shallow(<StyledCowComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders animalChat', () => {
    const chat = true;
    const wrapper = shallow(<StyledCowComponent chat={chat} />);
    expect(wrapper).toMatchSnapshot();
  });
});
