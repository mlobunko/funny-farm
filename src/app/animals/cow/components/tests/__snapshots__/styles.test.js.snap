// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCatComponent renders animal 1`] = `
<div
  aria-label="play cow sound button"
  className="sc-bdVaJa jKWUBw"
/>
`;

exports[`StyledCatComponent renders animalChat 1`] = `
<div
  aria-label="play cow sound button"
  className="sc-bdVaJa WgoNZ"
/>
`;
